File: solutions/go/roman-numerals/1/roman_numerals.go
package romannumerals

import (
    "errors"
)

func ToRomanNumeral(input int) (string, error) {
    romanValues := []struct {
		Value  int
		Symbol string
	}{
		{1000, "M"},
		{900, "CM"},
		{500, "D"},
		{400, "CD"},
		{100, "C"},
		{90, "XC"},
		{50, "L"},
		{40, "XL"},
		{10, "X"},
		{9, "IX"},
		{5, "V"},
		{4, "IV"},
		{1, "I"},
	}
    if input <= 0 || input > 3999  {
        return "", errors.New("Input should be a positive number above zero.")
    }
    romanNum := ""
    for _, roman := range romanValues {
		for input >= roman.Value {
			romanNum += roman.Symbol
			input -= roman.Value
		}
	}
	return romanNum, nil
}

File: solutions/csharp/roman-numerals/1/RomanNumerals.cs
using System;
using System.Collections.Generic;
using System.Text;

public static class RomanNumeralExtension
{
    private static readonly List<KeyValuePair<int, string>> digitMap = new List<KeyValuePair<int, string>>
    {
        new KeyValuePair<int, string>(1000, "M"),
        new KeyValuePair<int, string>(900, "CM"),
        new KeyValuePair<int, string>(500, "D"),
        new KeyValuePair<int, string>(400, "CD"),
        new KeyValuePair<int, string>(100, "C"),
        new KeyValuePair<int, string>(90, "XC"),
        new KeyValuePair<int, string>(50, "L"),
        new KeyValuePair<int, string>(40, "XL"),
        new KeyValuePair<int, string>(10, "X"),
        new KeyValuePair<int, string>(9, "IX"),
        new KeyValuePair<int, string>(5, "V"),
        new KeyValuePair<int, string>(4, "IV"),
        new KeyValuePair<int, string>(1, "I"),
    };

    public static string ToRoman(this int value)
    {
        if (value <= 0 || value > 3999)
            throw new ArgumentException("Input should be a positive number above zero and less than 4000.");

        var result = new StringBuilder();

        foreach (var pair in digitMap)
        {
            while (value >= pair.Key)
            {
                result.Append(pair.Value);
                value -= pair.Key;
            }
        }

        return result.ToString();
    }
}